### DISCUSSION TITLE
Composite Action Practice Program Issue: Getting a error Command Not Found for goodbye.sh in action.yml file

### DISCUSSION THREAD

**Discussion by IndiaLong3618** [community/community#52760] [Programming Help]
### Body

I'm receiving this error: /home/runner/work/_temp/49fb992e-59ec-4833-b0e6-564e06bbcd91.sh: line 1: goodbye.sh: command not found
Error: Process completed with exit code 127.
I ran a couple commands such as 'ls -lR' to check the files checkout-ed in workspace, chmod to make file executable, and finally readlink -f to verify the file path and I'm still getting the same error.  - https://github.com/CVS-Enterprise-IT/india_Testing_Repo/actions/runs/4693559898/jobs/8320689616#step:18:23 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by airtower-luna** [comment]
It's possible that you're not calling the script correctly, if the directory is not in `PATH` you need to call it by its path, e.g. `./goodbye.sh`.

But that's just a guess, because that link gives me only a 404, probably the repository is private. If you need more help, please make sure to quote the relevant part of the workflow file.

**Reply by IndiaLong3618** [reply]
I'll copy the code. I originally put ./goodbye.sh and it still didn't work

**Reply by IndiaLong3618** [reply]
name: 'Hello World'
description: 'Greet someone'
inputs:
  who-to-greet:  # id of input
    description: 'Who to greet'
    required: true
    default: 'World'
outputs:
  random-number:
    description: "Random number"
    value: ${{ steps.random-number-generator.outputs.random-number }}
runs:
  using: "composite"
  steps:
    - run: echo Hello ${{ inputs.who-to-greet }}.
      shell: bash
    - id: random-number-generator
      run: echo "random-number=$(echo $RANDOM)" >> $GITHUB_OUTPUT
      shell: bash
    - run: echo "${{ github.action_path }}" >> $GITHUB_PATH
      shell: bash
    - run: ./india_Testing_Repo/goodbye.sh
      shell: bash

**Reply by IndiaLong3618** [reply]
on: [push]

jobs:
  hello_world_job:
    runs-on: ubuntu-latest
    name: A job to say hello
    steps:
      - uses: actions/checkout@v3
      - run: chmod +x goodbye.sh
        shell: bash
      - run: ls -lR
        shell: bash
      - run: readlink -f goodbye.sh
        shell: bash
      - id: foo
        uses: CVS-Enterprise-IT/india_Testing_Repo/.github/actions@v1
        with:
          who-to-greet: 'Neo the Matrix cat'
      - run: echo random-number ${{ steps.foo.outputs.random-number }}
        shell: bash

**Comment by IndiaLong3618** [comment]
folder paths 
- [india_Testing_Repo](https://github.com/CVS-Enterprise-IT/india_Testing_Repo)/[.github](https://github.com/CVS-Enterprise-IT/india_Testing_Repo/tree/main/.github)/[workflows](https://github.com/CVS-Enterprise-IT/india_Testing_Repo/tree/main/.github/workflows)/main.yml

**Comment by IndiaLong3618** [comment]
[india_Testing_Repo](https://github.com/CVS-Enterprise-IT/india_Testing_Repo)/[.github](https://github.com/CVS-Enterprise-IT/india_Testing_Repo/tree/main/.github)/[actions](https://github.com/CVS-Enterprise-IT/india_Testing_Repo/tree/main/.github/actions)/action.yml

**Comment by IndiaLong3618** [comment]
I originally thought that there was an issue calling the action where the workflow is in the same repo but I've gotten around that error. The only one consist is the "command not found" for the goodbye.sh

**Comment by airtower-luna** [comment]
When posting code, *always* use a code block (enclosed in triple backticks). In YAML indentation is part of the syntax, so a wrong space might break everything.

That said, where is the script file used in your composite action supposed to come from?

```yaml
 - run: ./india_Testing_Repo/goodbye.sh
   shell: bash
```

Is it in the action repository? In the repository where the workflow is?
